core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/Shape.java

```diff
@@ -430,7 +429,8 @@ public boolean requiresEvaluation(ConnectionsGroup connectionsGroup, Scope scope
 	 */
 	public SparqlFragment buildSparqlValidNodes_rsx_targetShape(StatementMatcher.Variable subject,
 			StatementMatcher.Variable object,
-			RdfsSubClassOfReasoner rdfsSubClassOfReasoner, Scope scope) {
+			RdfsSubClassOfReasoner rdfsSubClassOfReasoner, Scope scope,
+			StatementMatcher.StableRandomVariableProvider stableRandomVariableProvider) {
```

Comment:
Plan nodes that run SPARQL queries also need to support `.equals` and `.hashCode`. That only works if the code that generates the SPARQL queries generate the same query for every run (eg. functional). The `StatementMatcher.StableRandomVariableProvider` gives sequential variables prefixed by a static UUID. Name is a bit long I might admit.